File: typescript/examples/render.ts
import { ImpossibleFX } from '../src/index.js';

async function main() {
  const client = new ImpossibleFX({
    region: 'us-east-1',
    apiKey: process.env.IMPOSSIBLE_API_KEY,
  });

  // Render a movie with parameters
  const result = await client.render('my-project', 'intro-video', {
    title: 'Welcome',
    subtitle: 'Getting Started with Impossible FX',
    color: '#ff6600',
  }, {
    format: 'mp4',
  });

  console.log('Render complete!');
  console.log('Token:', result.token);
  console.log('URL:', result.url);
  console.log('Expires:', new Date(result.expires * 1000).toISOString());

  // Alternatively, create a token first and poll for progress
  const tokenResult = await client.createToken('my-project', 'intro-video', {
    title: 'Async Render',
  });

  console.log('\nToken created:', tokenResult.token);

  // Build the URL manually
  const url = client.getUrl(tokenResult.token, 'mp4');
  console.log('Output URL:', url);

  // Poll for progress
  let progress = await client.getProgress(tokenResult.token);
  while (progress.done < progress.total) {
    console.log(`Progress: ${progress.done}/${progress.total}`);
    await new Promise((resolve) => setTimeout(resolve, 1000));
    progress = await client.getProgress(tokenResult.token);
  }
  console.log('Render finished!');
}

main().catch(console.error);

File: rust/examples/render.rs
use std::collections::HashMap;

use impossible_fx_sdk::{ImpossibleFX, RenderOptions};

#[tokio::main]
async fn main() -> impossible_fx_sdk::Result<()> {
    // Create a client with an API key.
    let client = ImpossibleFX::builder("us-east-1")
        .api_key("your-api-key")
        .build()?;

    // Build render parameters.
    let mut params = HashMap::new();
    params.insert("title".to_string(), serde_json::json!("Hello, World!"));
    params.insert("color".to_string(), serde_json::json!("#ff0000"));

    // Render a movie with custom options.
    let result = client
        .render(
            "my-project",
            "intro",
            &params,
            Some(RenderOptions {
                format: Some("mp4".to_string()),
                parallel: Some(2),
                ..Default::default()
            }),
        )
        .await?;

    println!("Token:   {}", result.token);
    println!("URL:     {}", result.url);
    println!("Expires: {}", result.expires);
    println!("Status:  {}", result.status);

    // You can also create a token and poll for progress.
    let token_result = client
        .create_token("my-project", "intro", &params, None)
        .await?;

    let url = client.get_url(&token_result.token, "mp4");
    println!("\nToken URL: {}", url);

    let progress = client.get_progress(&token_result.token).await?;
    println!("Progress: {}/{}", progress.done, progress.total);

    Ok(())
}
